artifactory/commands/generic/upload.go

```diff
@@ -271,3 +279,30 @@ func createDeleteSpecForSync(deletePattern string, syncDeletesProp string) *spec
 		Recursive(true).
 		BuildSpec()
 }
+
+func recordUploadJobSummary(summary *rtServicesUtils.OperationSummary) (err error) {
+	jobSummary, err := jobsummaries.NewJobSummaryImpl(&jobsummariesimpl.GithubSummaryRtUploadImpl{})
+	if err != nil || jobSummary == nil {
+		return
+	}
+	if err = jobSummary.CreateSummaryMarkdown(readDetailsFromReader(summary.TransferDetailsReader), jobsummaries.ArtifactsUploadSection); err != nil {
+		return
+	}
+	return
+}
+
+// Reads transfer details from the reader and return the content as bytes for further processing
+func readDetailsFromReader(reader *content.ContentReader) []byte {
+	var readContent []byte
+	if reader != nil {
+		for _, file := range reader.GetFilesPaths() {
+			// Read source file
+			sourceBytes, err := os.ReadFile(file)
+			if err != nil {
+				return nil
```

Comment:
Looks like a bug here - should return an error, not nil.
We should also have CheckError here.